packages/eas-cli/src/commands/secret/sync.ts

```diff
@@ -0,0 +1,224 @@
+import { Errors, Flags } from '@oclif/core';
+import assert from 'assert';
+import chalk from 'chalk';
+import dotenv from 'dotenv';
+import fs from 'fs-extra';
+import path from 'path';
+
+import EasCommand from '../../commandUtils/EasCommand';
+import { ExpoGraphqlClient } from '../../commandUtils/context/contextUtils/createGraphqlClient';
+import { EASNonInteractiveFlag } from '../../commandUtils/flags';
+import { EnvironmentSecretFragment, EnvironmentSecretType } from '../../graphql/generated';
+import { EnvironmentSecretMutation } from '../../graphql/mutations/EnvironmentSecretMutation';
+import {
+  EnvironmentSecretScope,
+  EnvironmentSecretsQuery,
+} from '../../graphql/queries/EnvironmentSecretsQuery';
+import Log from '../../log';
+import { ora } from '../../ora';
+import {
+  getDisplayNameForProjectIdAsync,
+  getOwnerAccountForProjectIdAsync,
+} from '../../project/projectUtils';
+import { promptAsync } from '../../prompts';
+import intersection from '../../utils/expodash/intersection';
+
+export default class EnvironmentSecretSync extends EasCommand {
+  static override description = 'read environment secrets from env file and store on the server';
+
+  static override flags = {
+    scope: Flags.enum({
+      description: 'Scope for the secrets',
+      options: [EnvironmentSecretScope.ACCOUNT, EnvironmentSecretScope.PROJECT],
+      default: EnvironmentSecretScope.PROJECT,
+    }),
+    'env-file': Flags.string({
+      description: 'Env file with secrets',
+    }),
+    force: Flags.boolean({
+      description: 'Delete and recreate existing secrets',
+      default: false,
+    }),
+    ...EASNonInteractiveFlag,
+  };
+
+  static override contextDefinition = {
+    ...this.ContextOptions.ProjectConfig,
+    ...this.ContextOptions.LoggedIn,
+  };
+
+  async runAsync(): Promise<void> {
+    let {
+      flags: { scope, force, 'env-file': envFilePath, 'non-interactive': nonInteractive },
+    } = await this.parse(EnvironmentSecretSync);
+    const {
+      projectConfig: { projectId },
+      loggedIn: { graphqlClient },
+    } = await this.getContextAsync(EnvironmentSecretSync, {
+      nonInteractive,
+    });
+
+    const projectDisplayName = await getDisplayNameForProjectIdAsync(graphqlClient, projectId);
+    const ownerAccount = await getOwnerAccountForProjectIdAsync(graphqlClient, projectId);
+
+    if (!envFilePath) {
+      const validationMessage = 'Env file must be passed.';
+      if (nonInteractive) {
+        throw new Error(validationMessage);
+      }
+
+      ({ envFilePath } = await promptAsync({
```

Comment:
I'm not a fan of this pattern, consider extracting it so using let and that assert is not necessarry